### ISSUE TITLE
v1.7.8 breaks compiling document previously compiled with older version

### ISSUE THREAD

**Issue by francisco-dlp** [sphinx-doc/sphinx#5361]
### Problem
v1.7.8 fails compiling a document (in our case [the HyperSpy documentation](https://hyperspy.readthedocs.io/en/latest/)) that v1.7.7 compiles happily.

v1.7.7 works fine on the same document.

#### Procedure to reproduce the problem
```
make html
```

#### Error logs / results


```bash
reading sources... [  0%] api/hyperspy                                                                                                                                                                                                                                                    
Exception occurred:
  File "/home/user/anaconda/anaconda/lib/python3.6/site-packages/sphinx/environment/__init__.py", line 313, in clear_doc
    self.included.pop(docname, None)
TypeError: pop() takes no arguments (2 given)
The full traceback has been saved in /tmp/sphinx-err-q3vhqr4r.log, if you want to report the issue to the developers.
```

Full build error in ReadTheDocs: https://readthedocs.org/projects/hyperspy/builds/7701605/

### Reproducible project / your project
https://github.com/hyperspy/hyperspy/tree/FIX_RTD_apidoc


### Environment info

ReadTheDocs and local Anaconda installation running on Debian.

**Comment by TimKam** [comment]
You can try if deleting the previous build and rebuilding works as a quick fix. I had a similar issue after switching between two unreleased versions.

**Comment by francisco-dlp** [comment]
Thanks for the suggestion @TimKam. We experience this issue in ReadTheDocs where I believe that each build is performed in a clean virtual environment.

**Comment by TimKam** [comment]
@francisco-dlp RtD does not always trigger builds in a clean environment. Chances are you will fix the issue by "wiping" the version--which only removes files internally and does not take the hosted docs down--and triggering a rebuild manually. See: https://docs.readthedocs.io/en/latest/guides/wipe-environment.html. Please report back :-)

**Comment by TimKam** [comment]
@tk0miya This seems to be a bigger issue. The build of the latest Sphinx docs failed as well on RtD. I managed to build it by proceeding as described above.

**Comment by francisco-dlp** [comment]
Wiping the version in RtDs [worked](https://readthedocs.org/projects/hyperspy/builds/7702427/), thanks @TimKam !

**Comment by TimKam** [comment]
Thanks for confirming.
Issue _could_ have been caused by: https://github.com/sphinx-doc/sphinx/issues/5327.

**Comment by tk0miya** [comment]
OMG... Sorry for inconvenience.

It would be raised on incremental build just after upgrade. I suppose it gets errored only once.
But I agree this is major bug. So I'll work for this tonight (in JST).

**Comment by tk0miya** [comment]
Fixed by #5363.
Thank you for reporting.
I'll release this later.
